wallet/wallet.h

```diff
@@ -336,4 +337,18 @@ bool wallet_invoices_load(struct wallet *wallet, struct invoices *invs);
  */
 bool wallet_invoice_remove(struct wallet *wallet, struct invoice *inv);
 
+/**
+ * wallet_htlc_stubs - Retrieve HTLC stubs for the given channel
+ *
+ * Load minimal necessary information about HTLCs for the on-chain
+ * settlement. This returns a `tal_arr` allocated off of @ctx with the
+ * necessary size to hold all HTLCs.
+ *
+ * @ctx: Allocation context for the return value
+ * @wallet: Wallet to load from
+ * @chan: Channel to fetch stubs for
+ */
+struct htlc_stub *wallet_htlc_stubs(tal_t *ctx, struct wallet *wallet,
+				    struct wallet_channel *chan);
+
```

Comment:
So, we are wasting a lot of space in the DB by not cleaning up "channel_htlcs".  The save_htlc_stubs() calls were carefully placed for having a simple "htlc_stubs" table in the db; we'll have to dig that code out again when we optimize this, which is a bit annoying.

Still, a separate table is overkill for the moment and can certainly be added later with a simple db transform...

Reply:
We can probably stubbify by issuing an update that sets all non-stub fields to `null`, that'd be almost as small as the separate table (especially when dropping the onion).